reconcile/utils/ocm/status_board.py

```diff
@@ -17,7 +17,7 @@ class BaseOCMSpec(TypedDict):
 
 
 class ApplicationOCMSpec(BaseOCMSpec):
-    product_id: str
+    product: dict[str, str]
```

Comment:
please create a new TypedDict class to enforce `id` field, can use a generic one for both `product` and `application`

```python
class IDSpec(TypedDict):
    id: str
```